js/pulse.js

```diff
@@ -216,18 +223,20 @@ var PulseMaker = {
       }
 
       if(row > 1) {
-        if(!newData[row]) {
-          newData[row] = {};
+        if(!projects[row]) {
+          projects[row] = {};
         }
-        newData[row][headings[col]] = value;
+        projects[row][headings[col]] = value;
       }
     }
 
-    for(var k in newData){
-      finalData.push(newData[k]);
+    for(var rowNum in projects){
+      var projectData = projects[rowNum];
+      projectData.id = this.generatePulseProjectId(projectData); 
```

Comment:
changes in `parseGoogleSheetData` method are mostly var ranaming except this one - adding `id` property to project